Tan_ShinYi/3. TDD/SLL/sll.rb

```diff
@@ -0,0 +1,133 @@
+class SLLNode
+  attr_reader :val
+  attr_accessor :next
+  def initialize value
+    @val = value
+    @next = nil
+  end
+end
+
+class SLL
+  attr_accessor :head, :tail
+  attr_reader :all_nodes, :all_values, :min, :max
+  def initialize
+    @head= nil
+    @tail=nil
+    @all_nodes=[]
+    @all_values=[]
+  end
+
+  def update
+    @all_nodes = []
+    @all_values = []
+    current = self.head
+    while current do
+      @all_nodes.push(current)
+      @all_values.push(current.val)
+      current=current.next
+    end
+  end
+
+  def insert val, val_after
+    count=1
+    current = self.head
+    while current do
+      if current.val==val_after
+        temp=current.next
+        current.next = SLLNode.new(val)
+        current.next.next = temp
+        if temp = nil
+          @tail=current.next
+        end
+        @all_nodes.insert(count, current.next)
+        @all_values.insert(count, current.next.val)
```

Comment:
this is pretty cool.  I like how you're keeping track of all_nodes and all_values and updating them